Formula/shtools.rb

```diff
@@ -0,0 +1,44 @@
+class Shtools < Formula
+  desc "Spherical Harmonic Tools"
+  homepage "https://shtools.github.io/SHTOOLS/"
+  url "https://github.com/SHTOOLS/SHTOOLS/archive/v4.7.tar.gz"
+  sha256 "2d6f426dbdeb8e3a16b1cd5ea5dd5db228bdb76f438abebcccd15e43096e8d37"
+  license "BSD-3-Clause"
+  head "https://github.com/SHTOOLS/homebrew-shtools.git"
+
+  depends_on "fftw"
+  depends_on "gcc"
+
+  def install
+    ENV.deparallelize
+    system "make", "fortran"
+    system "make", "fortran-mp"
+
+    pkgshare.install "examples/fortran/", "examples/ExampleDataFiles/"
+    inreplace pkgshare/"fortran/Makefile", "../../lib", "/usr/local/lib"
+    inreplace pkgshare/"fortran/Makefile", "../../modules", "/usr/local/include"
+
+    lib.install "lib/libSHTOOLS.a", "lib/libSHTOOLS-mp.a"
+    include.install "modules/fftw3.mod", "modules/planetsconstants.mod", "modules/shtools.mod", "modules/ftypes.mod"
+    share.install "man"
+  end
+
+  def caveats
+    <<~EOS
+      To use SHTOOLS with your gfortran code, compile with the options
```

Comment:
Is this specific to homebrew?

Reply:
No, these are the options that need to be included when you compile and link your code to `libSHTOOLS.a`.

Reply:
In that case it should be in the upstream documentation, not in caveats.

Reply:
Even though these instructions are found in the web documentation, a common question we always get is "what compiler flags do I need to use to link to libSHTOOLS.a and make it work?". If you don't think that this is appropriate, I won't complain too much about removing it.

Reply:
It's really just for issues that are specific to the homebrew install (https://docs.brew.sh/Formula-Cookbook#caveats)

Reply:
Would it be ok to note that the compiled modules and library are located in `/usr/local/include` and `/usr/local/lib`, respectively? This would be in contrast to `/opt/local/...` when using macports. If not, I will just remove the entire caveats block.

Reply:
That's pretty standard for homebrew so I wouldn't bother.